js/ui/speed_dial_action/speed_dial_main_item.js

```diff
@@ -87,30 +87,37 @@ const SpeedDialMainItem = SpeedDialItem.inherit({
     },
 
     _renderClick() {
-        this._clickAction = this.option("actions").length === 1 ?
+        this._clickAction = this._getVisibleActions().length === 1 ?
             this._createActionByOption("onClick") :
             this._createAction(this._clickHandler);
 
         this._setClickAction();
     },
 
     _defaultActionArgs() {
-        const actions = this.option("actions");
+        const actions = this._getVisibleActions();
 
         return {
             component: actions.length === 1 ? actions[0] : this
         };
     },
 
+    _getVisibleActions(actions) {
+        const currentActions = actions || this.option("actions");
+
+        return currentActions.filter((action) => action.option("visible"));
+    },
+
     _clickHandler() {
-        const actions = this._actionItems;
-        actions.forEach(action => {
-            action.toggle();
+        const actions = this._actionItems.filter((action) => action.option("actionVisible"));
```

Comment:
Why we can't use 'visible' option and should save and then use additional 'actionVisible' option?

Reply:
Aren't they equal? You call `action._options.actionVisible = action._options.visible;` on render...